src/game/rtsutils.h

```diff
@@ -58,14 +59,14 @@ template<typename T> struct less_than_nocase
     bool operator()(const T &left, const T &right) const { return (left.Compare_No_Case(right) < 0); }
 };
 
-template<typename T> struct equal_to
+template<typename T> struct hash
 {
-    bool operator()(const T &left, const T &right) const { return (left.Compare(right) == 0); }
+    size_t operator()(const T &object) const { return static_cast<size_t>(object); }
 };
 
-template<typename T> struct hash
+template<> struct hash<Utf8String>
 {
-    size_t operator()(const T &object) const
+    size_t operator()(const Utf8String &object) const
```

Comment:
```c++
        do {
            hash = *c + 5 * hash;
            ++c;
        } while (*c != '\0');
```
This looks wrong. This will read string past the end if the first character is null.
The following makes more sense:
```c++
        while (*c != '\0') {
            hash = *c + 5 * hash;
            ++c;
        };
```

Reply:
While true that isn't what I was looking at as part of the pr tbh i would rather it just called the std library https://github.com/TheAssemblyArmada/Thyme/blob/f95ea5c66e3ecd31930f36883764f14d8d287cff/deps/stlport/stl/_hash_fun.h#L42

Reply:
We can do follow up change to tackle this problem indeed.